qcodes/utils/plotting.py

```diff
@@ -42,4 +67,180 @@ def auto_range_iqr(data_array: np.ndarray) -> Tuple[float, float]:
+        raise RuntimeError('Can only scale mesh data, but received '
+                           f'"{type(colorbar.mappable)}" instead')
+    if data_lim is None:
+        if data_array is None:
+            data_array = colorbar.mappable.get_array()
+        data_lim = np.nanmin(data_array), np.nanmax(data_array)
+    else:
+        if data_array is not None:
+            raise RuntimeError('You cannot specify `data_lim` and `data_array` '
+                               'please read the docstring of '
+                               '`apply_color_scale_limits:\n\n`'
+                               + apply_color_scale_limits.__doc__)
+        else:
+            data_lim = cast(Tuple[float, float], tuple(sorted(data_lim)))
+    # if `None` is provided in the new limits don't change this limit
+    vlim = [new or old for new, old in zip(new_lim, colorbar.get_clim())]
+    # sort limits in case they were given in a wrong order
+    vlim = sorted(vlim)
+    # detect exceeding colorscale and apply new limits
+    exceeds_min, exceeds_max = (data_lim[0] < vlim[0],
+                                data_lim[1] > vlim[1])
+    if exceeds_min and exceeds_max:
+        extend = 'both'
+    elif exceeds_min:
+        extend = 'min'
+    elif exceeds_max:
+        extend = 'max'
+    else:
+        extend = 'neither'
+    colorbar.extend = extend
+    # the colorbar has no setter method and setting the colorbar extend does
+    # not take any effect. Calling a subsequent update will cause a runtime
+    # error because of the internal implementation of the rendering of the
+    # colorbar. To circumvent this we need to manually specify the property
+    # `_inside`, which is a slice that describes which of the colorbar levels
+    # lie inside of the box and it is thereby dependent on the extend.
+    colorbar._inside = colorbar._slice_dict[extend]
+    cmap = colorbar.mappable.get_cmap()
+    cmap.set_over(color_over)
+    cmap.set_under(color_under)
+    colorbar.mappable.set_clim(vlim)
+
+
+def apply_auto_color_scale(colorbar: matplotlib.pyplot.colorbar,
+                           data_array: Optional[np.ndarray]=None,
+                           cutoff_percentile: Optional[Union[Tuple[
+                               Number, Number], Number]]=DEFAULT_PERCENTILE,
+                           color_over: Optional[Any]=DEFAULT_COLOR_OVER,
+                           color_under: Optional[Any]=DEFAULT_COLOR_UNDER
+                           ) -> None:
+    """
+    Sets the color limits such that outliers are disregarded.
+
+    This method combines the automatic color scaling from
+    :meth:`auto_range_iqr` with the color bar setting from
+    :meth:`apply_color_scale_limits`.
+    If you want to adjust the color scale based on the configuration file
+    `qcodesrc.json`, use :meth:`auto_color_scale_from_config`, which is used
+    In :meth:`plot_by_id` and `show_num` (in qdev-wrappers),
```

Comment:
i don't think we should refer to functions from qdev-wrappers. this reference just does not seem to belong here.

Reply:
Also, i do not think that mentioning `show_num` in the example notebook is appropriate. `show_num` is known by a relatively specific group of users, while qcodes in general is targeting a broader range of users. It seems more fair to me to say smth like "..and custom plotting solutions" instead of mentioning `show_num`.